src/lighteval/metrics/__init__.py

```diff
@@ -120,15 +120,14 @@ def apply_generative_metric(
 
 def apply_multichoice_metric(results: list[ModelReturn], formatted_doc: Doc, metrics: list[str]):
     outputs = {}
-    if len(formatted_doc.choices) != len(results):
-        raise ValueError("Length of results is not equal to the length of the choices")
+    mc_results = results[: len(formatted_doc.choices)]
```

Comment:
To be clear, this is used for tasks that have both `loglikelihood` and `generative` metrics ?

Reply:
Yes exactly.